test/integration/replicaset/replicaset_test.go

```diff
@@ -515,3 +519,682 @@ func TestRSSelectorImmutability(t *testing.T) {
+	rs := newRS("rs", ns.Name, 2)
+	rs.Spec.Selector.MatchLabels = labelMap
+	rs.Spec.Template.Labels = labelMap
+
+	// Create 2 pods (one is failed, one is being deleted)
+	// Ensure both are being created again in the end
+	failedPod := newMatchingPod("failed-pod", ns.Name)
+	failedPod.Labels = labelMap
+	failedPod.Status.Phase = v1.PodFailed
+	deletedPod := newMatchingPod("deleted-pod", ns.Name)
+	deletedPod.Labels = labelMap
+	deletedPod.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+	createRSsPods(t, c, []*v1beta1.ReplicaSet{rs}, []*v1.Pod{failedPod, deletedPod}, ns.Name)
+
+	stopCh := make(chan struct{})
+	informers.Start(stopCh)
+	go rm.Run(5, stopCh)
+	defer close(stopCh)
+	waitRSStable(t, c, rs, ns.Name)
+
+	rs, err := c.ExtensionsV1beta1().ReplicaSets(ns.Name).Get(rs.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get replicaset %s: %v", rs.Name, err)
+	}
+	if rs.Status.Replicas != 2 {
+		t.Fatalf("Unexpected .Status.Replicas: Expected 2, saw %d\n", rs.Status.Replicas)
+	}
+
+	// Update .Spec.Replicas to 3 > 2
+	*(rs.Spec.Replicas) = 3
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Update(rs)
+	if err != nil {
+		t.Fatalf("failed to update replicaset %s: %v", rs.Name, err)
+	}
+	waitRSStable(t, c, rs, ns.Name)
+
+	// After updating the RS and waiting RS to be stable, we still have to GET the RS
+	// for the RS to reflect newest .Status
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Get(rs.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get replicaset %s: %v", rs.Name, err)
+	}
+	if rs.Status.Replicas != 3 {
+		t.Fatalf("Unexpected .Status.Replicas: Expected 3, saw %d\n", rs.Status.Replicas)
+	}
+
+	// Update .Spec.Replicas to 2 < 3
+	*(rs.Spec.Replicas) = 2
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Update(rs)
+	if err != nil {
+		t.Fatalf("failed to update replicaset %s: %v", rs.Name, err)
+	}
+	waitRSStable(t, c, rs, ns.Name)
+
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Get(rs.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get replicaset %s: %v", rs.Name, err)
+	}
+	if rs.Status.Replicas != 2 {
+		t.Fatalf("Unexpected .Status.Replicas: Expected 2, saw %d\n", rs.Status.Replicas)
+	}
+
+	// Update .Spec.Replicas to 0
+	*(rs.Spec.Replicas) = 0
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Update(rs)
+	if err != nil {
+		t.Fatalf("failed to update replicaset %s: %v", rs.Name, err)
+	}
+	waitRSStable(t, c, rs, ns.Name)
+
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Get(rs.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get replicaset %s: %v", rs.Name, err)
+	}
+	if rs.Status.Replicas != 0 {
+		t.Fatalf("Unexpected .Status.Replicas: Expected 0, saw %d\n", rs.Status.Replicas)
+	}
+
+	// RS should reconcile conflicting attributes of .spec and .status
+	*(rs.Spec.Replicas) = 1
+	rs.Status.FullyLabeledReplicas = 10
+	rs, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).Update(rs)
```

Comment:
You can't use Update() to edit Status. The change will be ignored. What exactly are you trying to test here?

Reply:
I wish to test whether RS can reconcile conflicting attributes of `.spec` and `.status`. This may be unnecessary.

Reply:
Is this an attempt to translate an existing test? If so, we should figure out what that test was really checking.

As a rule of thumb, no controller integration test should involve the test code directly editing Status of the controller under test. The controller itself is solely responsible for updating its own Status.

Reply:
It is this test that i want to convert, but the test may be more suited as a unit test: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set_test.go#L354